### ISSUE TITLE
Automatically truncate long table names or throw useful error

### ISSUE THREAD

**Issue by dmarcelino** [balderdashy/sails-mysql#197]
As outlined in https://github.com/balderdashy/waterline/issues/725 (and balderdashy/waterline#497), this adapter does not deal with the 64 character limit on table names.  This is often an issue with the automatically-named junction tables.  Part of this fix may be to create a convention for mapping the table names that waterline-schema creates to a new, shorter name (used only internally to the adapter).  Also, bear in mind that it seems to be part of the WL roadmap to support the custom of junction table names (balderdashy/waterline-schema#20), in the case that this adapter would like to put the onus on the user to create properly-named junction tables.

In either case, this adapter needs to deal with this by choosing shorter table names or throwing useful errors.  Documentation and tests should accompany these updates.

**Comment by sailsbot** [comment]
Thanks for posting, @dmarcelino.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
